L108-node-grpc-reflection-library.md

```diff
@@ -0,0 +1,115 @@
+Node.js Reflection Server Library
+----
+* Author(s): jtimmons
+* Approver: murgatroid99
+* Status: In Review
+* Implemented in: Node.js
+* Last updated: 2023-10-10
+* Discussion at: https://groups.google.com/g/grpc-io/c/Ie2jFIHCwrc
+
+## Abstract
+
+Create a canonical implementation of the gRPC reflection API for Node.js based on the logic from the [nestjs-grpc-reflection library](https://gitlab.com/jtimmons/nestjs-grpc-reflection-module/-/blob/30b67a78ff99e31ae54a0ab34c3784316579c665/src/grpc-reflection/grpc-reflection.service.ts)
+
+## Background
+
+Since its introduction in 2017 there have been a variety of external node.js implementations for the [gRPC Reflection Specification](https://github.com/grpc/grpc/blob/ce75ec23a1a9c5239834b92da4ce0992d367a39c/doc/server-reflection.md), each of which is in various states of maintenance. A few examples are linked at the bottom of this section.
+
+This feature was initially requested in grpc/grpc-node#79
+
+* https://gitlab.com/jtimmons/nestjs-grpc-reflection-module
+* https://github.com/deeplay-io/nice-grpc/tree/master/packages/nice-grpc-server-reflection
+* https://github.com/AckeeCZ/grpc-server-reflection
+
+### Related Proposals:
+* Initial reflection proposal: https://github.com/grpc/proposal/blob/master/A15-promote-reflection.md
+* Proposal of API for similar library: https://github.com/grpc/proposal/blob/master/L106-node-heath-check-library.md
+
+## Proposal
+
+We are proposing the creation of a new `@grpc/reflection-server` package with the following external interface:
```

Comment:
If we ever want to provide reflection client functionality in the future, I think we would want it in the same package, so I would prefer to call the package `@grpc/reflection`.

Reply:
makes sense, will rename 👍